### ISSUE TITLE
contract(A, B, C) silently returns 0 when A and B share no labels (multi-input symmetric contractor bug)

### ISSUE THREAD

**Issue by yingjerkao** [tenax-lab/tenax#553]
## Summary

`contract(*tensors)` on `SymmetricTensor` silently returns a zero tensor when given **three or more inputs** in a topology where some pair of inputs shares no labels (i.e. an intermediate step would require a tensor product). The bug reproduces on all symmetry groups (U(1), Z_n, FermionParity) and is independent of whether the tensors are fermionic.

The same network contracted pairwise via repeated 2-input `contract` calls gives the correct answer.

## Minimal reproducer

```python
import os
os.environ[\"JAX_PLATFORMS\"] = \"cpu\"

import jax
import numpy as np
from tenax.contraction.contractor import contract
from tenax.core.index import FlowDirection, TensorIndex
from tenax.core.symmetry import U1Symmetry
from tenax.core.tensor import SymmetricTensor


def make(seed, labels, flows):
    sym = U1Symmetry()
    indices = tuple(
        TensorIndex.from_charges(sym, np.array([0, 0], dtype=np.int32), f, label=l)
        for l, f in zip(labels, flows)
    )
    return SymmetricTensor.random_normal(indices, jax.random.PRNGKey(seed))


# Topology: A and B don't share any labels; both share separately with C.
A = make(1, (\"a\",),       (FlowDirection.IN,))
B = make(2, (\"b\",),       (FlowDirection.IN,))
C = make(3, (\"a\", \"b\"), (FlowDirection.OUT, FlowDirection.OUT))

print(\"3-input :\", float(contract(A, B, C).todense()))          # 0.0  <-- BUG
print(\"pairwise:\", float(contract(contract(A, B), C).todense()))  # correct (e.g. 0.93...)
print(\"pairwise:\", float(contract(contract(A, C), B).todense()))  # correct (e.g. 0.93...)
```

Output:
```
3-input : 0.0
pairwise: 0.9319159396418342
pairwise: 0.9319159396418343
```

The 2-input tensor product `contract(A, B)` (no shared labels) works correctly and produces the right outer product — the bug only appears when a third input is added.

## Reproduces on

- `U1Symmetry()` ✓
- `ZnSymmetry(2)` ✓
- `FermionParity()` ✓

## Cross-check: structurally similar topology that works

The triangular closure `T1=(a,b), T2=(b,c), T3=(c,a)` — where every pair shares one label — gives the correct answer in 3-input form. So the bug is specifically about cases where some pair of inputs is **disjoint** in labels.

## Why this matters

Encountered while building a fermionic-PEPS reference for #392 (`tests/test_fermionic_ed_reference.py`). The energy on a 2x2 PBC torus,

```python
contract(top_row, bot_row, G)
```

where `G` is the 2-site Hamiltonian gate (shares phys labels with `top_row`) and `bot_row` (shares chi labels with `top_row` but nothing with `G`), silently returned 0.0 for **every** gate and every observable insertion. The pairwise version `contract(contract(top_row, G), bot_row)` returned the correct value (e.g. -8.7e-4).

This means any caller that uses 3+ inputs to `contract()` for fermionic / symmetric PEPS observables — including normal usage like \"network · gate\" insertions — is at risk of silent zero results without any warning.

## Likely culprit

`_contract_symmetric` in `src/tenax/contraction/contractor.py`. The `common_sigs` computation (find contracted-charge signatures shared across all tensors) intersects sig keys from each tensor, but when one tensor contributes no contracted labels to a particular sig partition (because it shares no contracted labels with another tensor), its sig is the empty tuple, while the others' sigs are non-empty — and the intersection is empty, so no blocks are produced.

```python
# contractor.py L466-472
if tensor_indices_by_sig:
    common_sigs = set(tensor_indices_by_sig[0].keys())
    for idx_map in tensor_indices_by_sig[1:]:
        common_sigs &= set(idx_map.keys())
else:
    common_sigs = set()
```

When B has no contracted legs with A (only with C), the canonical-order signature for B is over a different label set than A's, and the intersection collapses to `{}`, returning zero output blocks.

## Suggested fix

Two options:

1. **Force pairwise execution under the hood**: when `len(tensors) >= 3` and the contraction graph has a disjoint pair, the multi-input contractor should fall back to opt_einsum's chosen pairwise ordering instead of trying to do everything in one block-pairing pass.
2. **Generalize the sig structure** to permit each tensor to have its own contracted-label set, and iterate over the cross-product of (per-tensor sigs) consistent with the global label graph — more invasive.

(1) is the safer fix; (2) would let the cuTENSOR path stay multi-input but is a real refactor.

## Workaround

Use strict pairwise `contract(a, b)` calls and chain manually. Document the bug at any callsite that currently uses 3+ inputs on `SymmetricTensor`.

## Blocks

#392 (fermionic split-CTM convention work) — the ED reference scaffolding needs `contract(top_row, bot_row, G)` to work, currently routed through pairwise as a workaround.
